### PR TITLE
Adds a notification area that displays messages from the backend

### LINKED ISSUE #332: UI Widget signifying problems and asking for user action
## Abstract

We need a UI widget that should appear on the top right and should tell the user that something is wrong and action is required.

## Motivation

Some times user something is constantly wrong and a simple pop-up won't help solve the problem. For example as mentioned in [this](https://github.com/rotkehlchenio/rotkehlchen/issues/329) issue the upgrade of the DB after we maintain our own list of tokens can end up with unknown tokens in the DB that the user will need to report as a bug so that they can be identified or properly transformed.

## Specification

A widget on the top right that will have the triangle with the exclamation mark to let the user know he needs to take action because something is not all right. Hovering over it should display the list of warnings/errors that need action.

Each one of them should be dismissable. 

The widget should also be some sort of a double ended queue data structure. If new warnings/errors are seen in the backend they will override the old ones. Max messages number should be probably set to 10 or 15, depending on how cluttered the widget looks like.

### PR DESCRIPTION
**Pull request by kelsos**
Closes #332

### CODE REVIEW

**Comment by kelsos** [comment]
@LefterisJP could you have a look, and see if it works for you and if you think some functionality is missing?

I have no idea how I could actually write an integration test for this though.

**Comment by LefterisJP** [comment]
Solved conflict and rebased on master

File: ui/utils.ts [hunk 1]
```diff
@@ -125,6 +132,43 @@ export function setup_client_auditor() {
...
+            notificationManager.mergeToCache(errors.concat(warnings));
+            updateNotifications();
+            setupNotificationHandlers();
+        })
+            .catch(reason => {
+                console.error(reason);
+            });
+    };
+    consume();
+    monitorMessages = setInterval(consume, 15000);
```
Comment (LefterisJP) [inline ui/utils.ts]: I guess every 10 seconds would even be better. It's a super cheap RPC call and the user should get these notification asap. Do you think 10 seconds is fine or even more frequent?

File: ui/notifications.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+}
+export function setupNotificationHandlers() {
+    const $notification = $('.notification');
+
+    const $clear = $('#notifications-clear-all');
+    $notification.off('click');
+
+    $clear.off('click');
+
+    $notification.on('click', function (event: JQuery.Event) {
```
Comment (LefterisJP) [inline ui/notifications.ts]: 2019-05-20-172722_339x426_scrot

So I have some comments here.

1. I see that the notification is being dismissed when it's clicked. I would rather there is a specific button. Like an [X] on the top right of each notification that dismisses it. Clicking anywhere else should not dismiss it. Perhaps there should also be a tooltip on top of that [X].
2. The garbage bin button on the top right is a bit too big. Should be a tiny tiny bit smaller. And most importantly it should:
    - have a tooltip that appears when hovering over it that will say that clicking will dismiss all open notification
    - have a confirmation menu. Clicking the button should not immediately remove all notifications but should ask you if you really want all active notifications to be removed.

File: ui/notifications.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        event.preventDefault();
+        $(event.target).closest('.notification').remove();
```
Comment (LefterisJP) [inline ui/notifications.ts]: Another thing that I just thought about is that with these notifications the old notifications (the bell thing should go away). I don't think anything is going to be appearing there anymore, right?

**Comment by LefterisJP** [comment]
Okay left some comments after testing it on my machine. As far as the tests are concerned, isn't there anyway to have a ready DB with something that would generate a warning, open rotkehlchen and see that they are there?

It could be as simple as a DB with an unknown asset.

File: ui/utils.ts (hunk 1 above)
Comment (kelsos) [reply ui/utils.ts]: If it doesn't cost much I don't see any reason why we can't go to 5 seconds.

File: ui/notifications.ts (hunk 2 above)
Comment (kelsos) [reply ui/notifications.ts]: Shouldn't be too much effort to update it. Thanks.

File: ui/notifications.ts (hunk 3 above)
Comment (kelsos) [reply ui/notifications.ts]: I can use the bell then for the new notifications and remove the old one?

Comment (kelsos) [reply ui/notifications.ts]: I really have no idea. I need to check when that part is updated, and maybe if there are any messages, integrate with the new system.

Comment (LefterisJP) [reply ui/notifications.ts]: I think it used to be that I checked the logs for warnings and put them there. But this was a bad idea and I stopped it. I think it should just go away.

File: ui/notifications.ts (hunk 2 above)
Comment (LefterisJP) [reply ui/notifications.ts]: Okay this is done. Tested it. Thanks.

**Comment by LefterisJP** [comment]
And ofcourse I forgot to push the changes I made to your branch .... which included a rebase and a typescript fix. No idea how it let me merge.

**Comment by kelsos** [comment]
If they didn't have any common changes maybe that is the reason. Now I have to do my scary rebase for the vue rewrite part.

**Pull request merged**